### PR TITLE
introduced _list fix

### PR DESCRIPTION
**Pull request by jkiddo**
Fixes https://github.com/hapifhir/hapi-fhir/issues/3761

### CODE REVIEW

**Comment by jkiddo** [comment]
ups ... Didn't intend to force you as reviewers. My bad.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java [hunk 1]
```diff
@@ -1985,8 +1986,11 @@ public IBundleProvider search(
 
 	private void translateListSearchParams(SearchParameterMap theParams) {
 
+		Set<Map.Entry<String, List<List<IQueryParameterType>>>> entryHashSet = new HashSet<>(theParams.entrySet());
+
 		// Translate _list=42 to _has=List:item:_id=42
-		for (String key : theParams.keySet()) {
+		for (Map.Entry<String, List<List<IQueryParameterType>>> stringListEntry : entryHashSet) {
+			String key = stringListEntry.getKey();
```
Comment (tadgh) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java:1993]: question: Why this change? It feels immaterial.

**Review by tadgh — APPROVED** [review]

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java (hunk 1 above)
Comment (jkiddo) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirResourceDao.java:1993]: See https://stackoverflow.com/a/16180833 - there's also a unit test that will fail if you revert that particular change.
